tests/functional/src/main/java/org/mule/functional/util/FlowExecutionLogger.java

```diff
@@ -27,21 +27,22 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 
 /**
  * Saves the payload that goes through a processor for accessing it later
  */
 public class FlowExecutionLogger extends AbstractComponent implements Processor {
 
   private static final int POLLING_TIMEOUT = 5000;
-  private static final int POLLING_DELAY = 0;
+  private static final int POLLING_DELAY = 500;
 
   private static final String EXECUTION_ROUTE_KEY = "executionRoute";
 
-  private static Map<String, ExecutionLog> executionLogsMap = new HashMap<>();
+  private static Map<String, ExecutionLog> executionLogsMap = new ConcurrentHashMap<>();
 
   public static void resetLogsMap() {
-    executionLogsMap = new HashMap<>();
+    executionLogsMap = new ConcurrentHashMap<>();
```

Comment:
why was this change needed?

Reply:
The aggregation-complete and aggregator-listener were executed concurrently and both were writing on the same at the same time.